### ISSUE TITLE
PrestaShop debug profiler doesn't work with Symfony layout

### ISSUE THREAD

**Issue by kpodemski** [PrestaShop/PrestaShop#35473]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

If you enable Symfony layout for the back office in feature flags, the built-in profiler doesn't show up.

### Expected behavior

Profiler should work as before because otherwise, it is impossible to easily debug the performance of queries coming from legacy, etc.

### Steps to reproduce

1. Enable Symfony layout in feature flags.
2. Edit config/defines.inc.php and change `_PS_DEBUG_PROFILING_` to `true`
3. See that the profiler doesn't show with Symfony layout, but it is available if Symfony layout is disabled.

### PrestaShop version(s) where the bug happened

9.0.0

### PHP version(s) where the bug happened

8.1

**Comment by florine2623** [comment]
Hello @kpodemski ,

I reproduce the issue with PrestaShop version 9.0.x, I'll add this to the backlog so it can be fixed.

https://github.com/PrestaShop/PrestaShop/assets/16019289/5be3870e-57b1-4ba8-aad6-e9f59f27da30

Please be aware that some issues might take a very long time to be resolved. If this one is important to you and you cannot wait for it to be fixed on the project’s own time, we strongly suggest you consider [contacting a professional](https://www.prestashop-project.org/support/) to help you. 

If you fix the issue on your end, please [contribute it back to the project](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thank you

**Comment by Hlavtox** [comment]
@kpodemski The profiler does not work at all on 9.0.0. Even if I disable symfony layout, it doesn't work - on legacy, on modern pages. Works only in FO.

Ping @jolelievre

**Comment by jolelievre** [comment]
Hi @kpodemski @Hlavtox 

yep the fact that the legacy debugger is not visible in 9.0 is wanted because all pages are now handled by Symfony and we already have the Symfony debug toolbar. I don't think having two different debug bar is a good idea.

HOWEVER, if there are some missing data present in the legacy debugger that are not visible with the Symfony one then the Symfony debug toolbar should be improved to cover these missing features (when relevant).

TLDR; we should add a data collector in the Symfony debug toolbar to display the legacy queries.

**Comment by kpodemski** [comment]
Hi @jolelievre 

Symfon debug toolbar only works with Doctrine and not the legacy database layer, that's why it was always required to have both enabled to better understand the performance impact.

**Comment by Hlavtox** [comment]
### Quick check Profiler vs symfony toolbar

### Missing
🔴Load Time 97.723 ms
🔴MySQL Version 10.6.16-MariaDB-0ubuntu0.22.04.1-log 
🔴Smarty Cache disabled 
🔴Smarty Compilation force compile
🔴Timing and memory usage of hooks
🔴Timing and memory usage of modules

### Partially missing
🟡Querying Time 11 ms -  legacy queries missing
🟡Queries 69 -  legacy queries missing
🟡Memory Limit	512M - missing - need to click full php config
🟡Max Execution Time - 600s -  missing - need to click full php config

### OK
🟢Memory Peak Usage 4.2 Mb
🟢Included Files 646 files - 8.11 Mb
🟢PrestaShop Version 8.1.7
🟢PHP Version 8.3.17
🟢loads ands loads of new useful stuff that was not in the profiler

**Comment by elyesnaouar** [comment]
@ibahloul-ps @jolelievre the issue leads us to false trails to compare between PS9 and old versions as the case for https://github.com/PrestaShop/PrestaShop/issues/39069

**Comment by Progi1984** [comment]
Fixed by #39621
